Add ROCm AMD GPU acceleration support (optional, detected at runtime)

geeksville opened the pull request:
This change increases the AI model processing speed >20x on my AMD GPU
equipped desktop.  A huge improvement in user experience.

Since I was building upon @adaloveseal 's other PR that added similar support for Intel GPUs, I went ahead and merged her branch into this.  

I also built this on top of the devcontainer support because dcs made it much easier to guarantee there is a properly configed developer environment.  If you ever merge this PR I'm happy to rebase however you wish.

Thank you for your work.

Now I'm starting work on a clean flatpak packing of this tool.

Commit:
feat: add OpenVINO runtime support for Intel Arc GPUs

geeksville committed:
Add support for devcontainers.

geeksville committed:
add missing requirements to fix bld

geeksville committed:
Merge remote-tracking branch 'adaloveseal/main'

geeksville committed:
better to only do the requirements install once, because it changes rarely

geeksville committed:
Install the python faulthandler code (even for non devs)

Graxpert can sometimes crash in native code.  And without the python
fault handler debugging such crashes is almost impossible.  Turn on the
fault handler so that if a crash occurs for an end-user it is more
likely they can send in at least stack traces.

Without this change a native crash log looks like:

```
Bus error (core dumped)
```

With this change a native crash log looks like:
```
vscode ➜ /workspaces/graxpert (main) $ python -X faulthandler -m graxpert.main
2025-07-08 18:39:28,113 MainProcess root INFO     /home/vscode/.config/GraXpert/preferences.json appears to be missing. it will be created after program shutdown
...
2025-07-08 18:39:33,268 MainProcess root WARNING  Could not check for newest version
Fatal Python error: Bus error

Thread 0x00007f81c71e16c0 (most recent call first):
  File "/usr/local/lib/python3.13/multiprocessing/connection.py", line 395 in _recv
  File "/usr/local/lib/python3.13/multiprocessing/connection.py", line 430 in _recv_bytes
  File "/usr/local/lib/python3.13/multiprocessing/connection.py", line 250 in recv
  File "/usr/local/lib/python3.13/multiprocessing/managers.py", line 831 in _callmethod
  File "<string>", line 2 in get
  File "/workspaces/graxpert/graxpert/mp_logging.py", line 82 in logger_thread
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap

Current thread 0x00007f821addbb80 (most recent call first):
  File "/workspaces/graxpert/graxpert/stretch.py", line 74 in stretch_all
  File "/workspaces/graxpert/graxpert/AstroImageRepository.py", line 85 in stretch_all
  File "/workspaces/graxpert/graxpert/application/app.py", line 560 in do_stretch
  File "/workspaces/graxpert/graxpert/application/app.py", line 549 in on_stretch_option_changed
  File "/workspaces/graxpert/graxpert/application/eventbus.py", line 21 in emit
  File "/workspaces/graxpert/graxpert/ui/statusbar.py", line 22 in <lambda>
  File "/usr/local/lib/python3.13/tkinter/__init__.py", line 2068 in __call__
  File "/usr/local/lib/python3.13/tkinter/__init__.py", line 427 in set
  File "/home/vscode/.local/lib/python3.13/site-packages/customtkinter/windows/widgets/ctk_optionmenu.py", line 381 in _dropdown_callback
  File "/home/vscode/.local/lib/python3.13/site-packages/customtkinter/windows/widgets/core_widget_classes/dropdown_menu.py", line 106 in _button_callback
  File "/home/vscode/.local/lib/python3.13/site-packages/customtkinter/windows/widgets/core_widget_classes/dropdown_menu.py", line 96 in <lambda>
  File "/usr/local/lib/python3.13/tkinter/__init__.py", line 2068 in __call__
  File "/usr/local/lib/python3.13/tkinter/__init__.py", line 1599 in mainloop
  File "/home/vscode/.local/lib/python3.13/site-packages/customtkinter/windows/ctk_tk.py", line 165 in mainloop
  File "/workspaces/graxpert/graxpert/main.py", line 186 in ui_main
  File "/workspaces/graxpert/graxpert/main.py", line 404 in main
  File "/workspaces/graxpert/graxpert/main.py", line 410 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, zstandard.backend_c, _cffi_backend, PIL._imaging, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, astropy.io.fits._utils, astropy.io.fits.hdu.compressed._compression, scipy._lib._ccallback_c, scipy.linalg._fblas, scipy.linalg._flapack, _cyutility, scipy._cyutility, scipy.linalg.cython_lapack, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, lz4._version, lz4.block._block, astropy.stats._stats, erfa.ufunc, astropy.stats._fast_sigma_clip, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, PIL._imagingmath, PIL._imagingtk (total: 85)
Bus error (core dumped)
```

geeksville committed:
Merge branch 'pr-devcont'

geeksville committed:
Merge branch 'pr-except'

geeksville committed:
Fix docker invocation so that python SharedMemory works correctly

geeksville committed:
Merge branch 'pr-devcont'

geeksville committed:
Add ROCm AMD GPU acceleration support (optional, detected at runtime)

This change increases the AI model processing >20x on my AMD GPU
equipped desktop.  A huge improvement in user experience.

schmelly commented:
@geeksville thanks for the pr. I stumbled upon one detail: so far, we install onnxruntime depending on the targeted platform in `.github/workflows/build-release.yml`. Thereby, I am not a fan of adding it to `requirements.txt`. I guess one way to support multiple platform/vendor combinations would be to add specific `requirements-<platform>-<vendor>.txt` files. What do you think?

CS, David

geeksville commented:
hmm.  Let me look a bit more as I am doing the 'containerization/appimage/linux-runtime cleanup' investigation and get back to you.  thank you for the comment!

geeksville committed:
Catch exceptions inside of GPU AI code, and fallback to CPU version

If the AI GPU code ever throws an exception (like it currently does for
the linux-exe build version - I'm working on that problem soon)
warn the user and fallback to the CPU based implementations.

Also clean up some related copypasta code that really should be shared
between all usages of AI (now in fallback_run())

schmelly commented:
@geeksville I guess we should synchronize on this pr once again: I guess the most straightforward way (for now?) to continue would be to create dedicated `graxpert-<platform>-<vendor>` packages due to onnxruntime dependencies. If you agree I would extend the build script appropriately.

geeksville commented:
@schmelly possibly!  I'm currently doing an experiment of using the pip variant dependency support (which is going pretty well).  I'll send you a link to a test build/branch in the next couple of days so you can play with it and see what you think.

This would allow stuff like "pip install graxpert[rocm]" or "pip install graxpert[gpu]" or "pip install [openvino]".  The default would be the "pip install[directml]" target.   This has a number of benefits:

* Each one of those subflavors will independently pullin whatver they need for that config. (and yeah sometimes the dependencies are large (though the user probably doesn't see this).  The nvidia "gpu" onnxruntime pulls down about 500MB, the rocm build pulls down about 900MB.
* Solves the problem (imo an oversight by the onnxruntime devs) of onnxruntime being 'fragile' and you can only have one set of executors in a python path at once - last one installed wins. 
* We could still distribute exe/msi installers, but for any system with "good/standard" pip support (windows, linux or mac) - users would have the option of "pip/pipx install graxpert[theirfavoriteflavor]" and the install would just work
* Because pypi/pip based dist is 'standard': Siril usage of graxpert becomes much easier because siril script writers could just add a dependency on the graxpert package and they would get **at least** a cpu capable version in their path.

cool?

papadako commented:
Any news on this? I would really appreciate having GraXper working on my AMD gpus!

geeksville commented:
It works great in my branch.  I'll send you a test build for feedback. The
new 7.0 rocm is VERY fast

(Sent from a phone - please ignore typos)

On Fri, Oct 3, 2025, 01:19 Papadakos Panagiotis ***@***.***>
wrote:

> *papadako* left a comment (Steffenhir/GraXpert#237)
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3362250907>
>
> Any news on this? I would really appreciate having GraXper working on my
> AMD gpus!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3362250907>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABXB2I357KL5EJHI6YKDMD3VVNCHAVCNFSM6AAAAACBLBVIFWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGNRSGI2TAOJQG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nostradukemas commented:
Is there a minimum spec of AMD GPU this works for? In the market for a new GPU on a Bazzite build and utilizing this would be a big check mark for me

geeksville commented:
In theory any AMD GPU will see some benefit.  I'm using a strix Halo 395 APU

(Sent from a phone - please ignore typos)

On Thu, Nov 20, 2025, 19:32 nostradukemas ***@***.***> wrote:

> *nostradukemas* left a comment (Steffenhir/GraXpert#237)
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3557506435>
>
> Is there a minimum spec of AMD GPU this works for? In the market for a new
> GPU on a Bazzite build and utilizing this would be a big check mark for me
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3557506435>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABXB2I2RLN7TACEIM7OMYD35WRDNAVCNFSM6AAAAACBLBVIFWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKNJXGUYDMNBTGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

geeksville commented:
But check AMD ROCm docs - I bet they have a list.

(Sent from a phone - please ignore typos)

On Fri, Nov 21, 2025, 12:14 Kevin Hester ***@***.***> wrote:

> In theory any AMD GPU will see some benefit.  I'm using a strix Halo 395
> APU
>
> (Sent from a phone - please ignore typos)
>
> On Thu, Nov 20, 2025, 19:32 nostradukemas ***@***.***>
> wrote:
>
>> *nostradukemas* left a comment (Steffenhir/GraXpert#237)
>> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3557506435>
>>
>> Is there a minimum spec of AMD GPU this works for? In the market for a
>> new GPU on a Bazzite build and utilizing this would be a big check mark for
>> me
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3557506435>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AABXB2I2RLN7TACEIM7OMYD35WRDNAVCNFSM6AAAAACBLBVIFWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKNJXGUYDMNBTGU>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

fliker09 commented:
OK, here is my attempt at running with 9060 XT:
1. Ubuntu 24.04
2. ROCm 7.1.1 (btw, your current version has a hard-coded 6.4.1, deffo worth looking into making it dynamic)
3. Graxpert installed from pip (3.2.0a2)
4. onnxruntime 7.0.2 (because 7.1.1 ditched ROCMExecutionProvider and left only MIGraphXExecutionProvider)

I added MIGraphXExecutionProvider to the list in ai_model_handling.py, but nevertheless, both it and ROCMExecutionProvider crash:
```
2026-01-08 01:21:28,729 MainProcess root INFO     Available inference providers : ['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']
2026-01-08 01:21:29.528235648 [W:onnxruntime:Default, migraphx_execution_provider.cc:180 get_flags_from_env] [MIGraphX EP] MIGraphX ENV Override Variables Set:
2026-01-08 01:21:34,980 MainProcess root INFO     Used inference providers for /home/alex/.local/share/GraXpert/bge-ai-models/1.0.1/model.onnx: ['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']
Fatal Python error: Segmentation fault

Current thread 0x00007c5cb0ae8140 (most recent call first):
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 273 in run
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 269 in <lambda>
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 228 in wrapped
  File "/usr/lib/python3.12/multiprocessing/process.py", line 108 in run
  File "/usr/lib/python3.12/multiprocessing/process.py", line 314 in _bootstrap
  File "/usr/lib/python3.12/multiprocessing/popen_fork.py", line 71 in _launch
  File "/usr/lib/python3.12/multiprocessing/popen_fork.py", line 19 in __init__
  File "/usr/lib/python3.12/multiprocessing/context.py", line 282 in _Popen
  File "/usr/lib/python3.12/multiprocessing/context.py", line 224 in _Popen
  File "/usr/lib/python3.12/multiprocessing/process.py", line 121 in start
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 234 in run_in_process
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 271 in __run_low
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 281 in run
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/background_extraction.py", line 74 in extract_background
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/application/app.py", line 168 in on_calculate_request
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/application/eventbus.py", line 21 in emit
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ui/left_menu.py", line 168 in <lambda>
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/customtkinter/windows/widgets/ctk_button.py", line 554 in _clicked
  File "/usr/lib/python3.12/tkinter/__init__.py", line 1967 in __call__
  File "/usr/lib/python3.12/tkinter/__init__.py", line 1504 in mainloop
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/customtkinter/windows/ctk_tk.py", line 165 in mainloop
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 210 in ui_main
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 445 in main_impl
  File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 487 in main
  File "/home/alex/Programs/Astronomy/GraXpert/bin/graxpert", line 7 in <module>

Extension modules: _cffi_backend, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, PIL._imaging, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, astropy.io.fits._utils, astropy.io.fits.hdu.compressed._compression, scipy._lib._ccallback_c, scipy.linalg._fblas, scipy.linalg._flapack, _cyutility, scipy._cyutility, scipy.linalg.cython_lapack, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._generator, numpy.random._mt19937, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, lz4._version, lz4.block._block, zstandard.backend_c, astropy.stats._stats, astropy.stats._fast_sigma_clip, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, PIL._imagingmath, PIL._imagingtk (total: 84)
2026-01-08 01:21:54,825 MainProcess root WARNING  Error running model, falling back to CPU only: worker process crashed with exit code -11.
```
I did try onnxruntime 7.1.1 as well, but it crashes all the same way, sadly. If this is not a proper place - I will move it to dedicated ticket!

geeksville commented:
No that's fine.  I'll try the latest onnxruntime with my AMD 395+ (AMD8060
APU/GPU).  At the very least I'll make sure that it fails over to CPU
implementation if ROCm blows where you had your crash.

On Thu, Jan 8, 2026 at 7:38 AM fliker09 ***@***.***> wrote:

> *fliker09* left a comment (Steffenhir/GraXpert#237)
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3721279355>
>
> OK, here is my attempt at running with 9060 XT:
>
>    1. Ubuntu 24.04
>    2. ROCm 7.1.1 (btw, your current version has a hard-coded 6.4.1, deffo
>    worth looking into making it dynamic)
>    3. Graxpert installed from pip (3.2.0a2)
>    4. onnxruntime 7.0.2 (because 7.1.1 ditched ROCMExecutionProvider and
>    left only MIGraphXExecutionProvider)
>
> I added MIGraphXExecutionProvider to the list in ai_model_handling.py, but
> nevertheless, both it and ROCMExecutionProvider crash:
>
> 2026-01-08 01:21:28,729 MainProcess root INFO     Available inference providers : ['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']
> 2026-01-08 01:21:29.528235648 [W:onnxruntime:Default, migraphx_execution_provider.cc:180 get_flags_from_env] [MIGraphX EP] MIGraphX ENV Override Variables Set:
> 2026-01-08 01:21:34,980 MainProcess root INFO     Used inference providers for /home/alex/.local/share/GraXpert/bge-ai-models/1.0.1/model.onnx: ['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']
> Fatal Python error: Segmentation fault
>
> Current thread 0x00007c5cb0ae8140 (most recent call first):
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 273 in run
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 269 in <lambda>
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 228 in wrapped
>   File "/usr/lib/python3.12/multiprocessing/process.py", line 108 in run
>   File "/usr/lib/python3.12/multiprocessing/process.py", line 314 in _bootstrap
>   File "/usr/lib/python3.12/multiprocessing/popen_fork.py", line 71 in _launch
>   File "/usr/lib/python3.12/multiprocessing/popen_fork.py", line 19 in __init__
>   File "/usr/lib/python3.12/multiprocessing/context.py", line 282 in _Popen
>   File "/usr/lib/python3.12/multiprocessing/context.py", line 224 in _Popen
>   File "/usr/lib/python3.12/multiprocessing/process.py", line 121 in start
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 234 in run_in_process
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 271 in __run_low
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ai_model_handling.py", line 281 in run
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/background_extraction.py", line 74 in extract_background
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/application/app.py", line 168 in on_calculate_request
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/application/eventbus.py", line 21 in emit
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/ui/left_menu.py", line 168 in <lambda>
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/customtkinter/windows/widgets/ctk_button.py", line 554 in _clicked
>   File "/usr/lib/python3.12/tkinter/__init__.py", line 1967 in __call__
>   File "/usr/lib/python3.12/tkinter/__init__.py", line 1504 in mainloop
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/customtkinter/windows/ctk_tk.py", line 165 in mainloop
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 210 in ui_main
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 445 in main_impl
>   File "/home/alex/Programs/Astronomy/GraXpert/lib/python3.12/site-packages/graxpert/main.py", line 487 in main
>   File "/home/alex/Programs/Astronomy/GraXpert/bin/graxpert", line 7 in <module>
>
> Extension modules: _cffi_backend, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, PIL._imaging, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, astropy.io.fits._utils, astropy.io.fits.hdu.compressed._compression, scipy._lib._ccallback_c, scipy.linalg._fblas, scipy.linalg._flapack, _cyutility, scipy._cyutility, scipy.linalg.cython_lapack, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._generator, numpy.random._mt19937, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, lz4._version, lz4.block._block, zstandard.backend_c, astropy.stats._stats, astropy.stats._fast_sigma_clip, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, PIL._imagingmath, PIL._imagingtk (total: 84)
> 2026-01-08 01:21:54,825 MainProcess root WARNING  Error running model, falling back to CPU only: worker process crashed with exit code -11.
>
> I did try onnxruntime 7.1.1 as well, but it crashes all the same way,
> sadly. If this is not a proper place - I will move it to dedicated ticket!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-3721279355>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABXB2M6KZ7UH6AMIUMHYZT4FWKGFAVCNFSM6AAAAACBLBVIFWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTOMRRGI3TSMZVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

fliker09 commented:
My bad, forgot to mention that it does fallback to CPU! But I want GPU acceleration, AI denoise is SUPER slow on CPU -_-

fliker09 commented:
AMD, why are you making it so hard?! I've installed ROCm 6.4.4 in parallel (doable, though a bit of a hassle). GraXpert still failing... BUT! Siril works with 6.4.4! And it's not using the onnxruntime wheel from AMD themselves... Removed the AMD version from GraXpert venv and installed the standard rocm one, to replicate the setup. Sadly, crashing with a new error:
```
2026-01-12 01:35:25,689 MainProcess root INFO     Used inference providers for /home/alex/.local/share/GraXpert/denoise-ai-models/3.0.2/model.onnx: ['ROCMExecutionProvider', 'CPUExecutionProvider']
2026-01-12 01:35:25,756 MainProcess root WARNING  Error running model, falling back to CPU only: [ONNXRuntimeError] : 1 : FAIL : /root/onnx_runtime/onnxruntime/onnxruntime-1.22.2/onnxruntime/core/providers/rocm/rocm_call.cc:141 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::RocmCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = hipError_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /root/onnx_runtime/onnxruntime/onnxruntime-1.22.2/onnxruntime/core/providers/rocm/rocm_call.cc:134 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::RocmCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = hipError_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] HIP failure 2: out of memory ; GPU=0 ; hostname=AMR5 ; file=/root/onnx_runtime/onnxruntime/onnxruntime-1.22.2/onnxruntime/core/providers/rocm/rocm_stream_handle.cc ; line=220 ; expr=hipStreamCreateWithFlags(&stream, hipStreamNonBlocking);
```
Interesting what makes Siril work and GraXpert not... Note - I've excluded MIGraphXExecutionProvider on both sides, as it crashes at all times, with both versions of the ROCm.

geeksville commented:
btw I'll be looking at this in the next few days.  Though I'm also trying to learn a new ergo keyboard layout (colmak-dh) and omg my typing speed is (temporarily?) so slow.  heh.

geeksville commented:
ok sorry - got distracted with my super slow typing and just practicing to type faster (and fixing a couple of bugs in the keyboard firmware).  I'll work on this next week (really).  sorry again!

geeksville commented:
alas an update: my typing is still slow and my occasionally shitty back is having one of its shitty flareups ;-).  So I'm extending my break from 'real' programming one more week.  But I promise this is at the top of my (fun!) programming queue...

geeksville commented:
Ok I (finally - sorry) looked into this.  With the latest 7.2 ROCm stuff I also see the same failure on my 395+ APU.  I think it **might** be the segfault they describe [here](https://rocm.docs.amd.com/projects/radeon-ryzen/en/latest/docs/limitations/limitationsrad.html#known-issues).  They say they are working on it. 

I think I'm just going to wait for 7.3 (which should be soon).  I have this changelist ready: https://github.com/geeksville/GraXpert/tree/feat-rocm-migraphx

Ilwyd commented:
Just curious, any plans on this being completed? Would love to be able to denoise in less than 30 minutes on my CPU

geeksville commented:
alas - ROCm releases are kinda variable.  I have a working version on my
machine which I've been happily using - I'll try again with a current
release and if it is fine push an update in a couple of weeks?

On Sun, Aug 9, 2026 at 1:36 PM Ilwyd ***@***.***> wrote:

> *Ilwyd* left a comment (Steffenhir/GraXpert#237)
> <https://github.com/Steffenhir/GraXpert/pull/237#issuecomment-5233720070>
>
> Just curious, any plans on this being completed? Would love to be able to
> denoise in less than 30 minutes on my CPU
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Steffenhir/GraXpert/pull/237?email_source=notifications&email_token=AABXB2KHABHDSDKSVDKFPRT5JDODFA5CNFSNUABFM5UWIORPF5TWS5BNNB2WEL2JONZXKZKDN5WW2ZLOOQXTKMRTGM3TEMBQG4YKM4TFMFZW63VHNVSW45DJN5XKKZLWMVXHJLDGN5XXIZLSL5RWY2LDNM#issuecomment-5233720070>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABXB2NGAUREMNHK6GH4FN35JDODFAVCNFSNUABFKJSXA33TNF2G64TZHM2DKOBYGMZDMNJWHNEXG43VMU5TGMRSGQ2TMMRQGY2KC5QC>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
